### PR TITLE
UCP/CONTEXT: incompatible version warning

### PR DESCRIPTION
**Pull request by hoopoepg**
- do not fail initialize on incompatible version
- updated gtests

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6256/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6258/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_context.cc [hunk 1]
```diff
@@ -79,13 +79,14 @@ UCS_TEST_P(test_ucp_version, wrong_api_version) {
...
     ucs_status_t status;
     {
-        wrap_errors();
+        hide_warnings();
         status = ucp_init_version(99, 99, &params, config.get(), &ucph);
         restore_errors();
     }
-    if (status == UCS_OK) {
+    if (status != UCS_OK) {
+        ADD_FAILURE() << "Failed to create UCP with wrong version";
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_context.cc]: need to also check that we had a warning (by a counter?)

File: test/gtest/common/test.cc [hunk 2]
```diff
@@ -175,6 +180,24 @@ test_base::hide_errors_logger(const char *file, unsigned line, const char *funct
...
 
+ucs_log_func_rc_t
+test_base::hide_warns_logger(const char *file, unsigned line, const char *function,
+                             ucs_log_level_t level, const char *message, va_list ap)
+{
+    if (level == UCS_LOG_LEVEL_WARN) {
+        pthread_mutex_lock(&m_logger_mutex);
+        va_list ap2;
+        va_copy(ap2, ap);
+        m_errors.push_back(format_message(message, ap2));
```
Comment (yosefe) [inline test/gtest/common/test.cc]: maybe add another vector m_warnings?

Comment (hoopoepg) [reply test/gtest/common/test.cc]: added

File: test/gtest/ucp/test_ucp_context.cc (hunk 1 above)
Comment (hoopoepg) [reply test/gtest/ucp/test_ucp_context.cc]: added warning count evaluation

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6266/ for details (Mellanox internal link).

**Pull request merged**